app/src/main/java/helper/CourseRequestHandler.java

```diff
@@ -149,10 +157,52 @@ public void onFailure(Call<List<CourseSection>> call, Throwable t) {
     }
 
 
+    private List<CourseSection> resolve(Object response) {
+        List<CourseSection> courseSections = (List<CourseSection>) response;
+        List<Content> contents = new ArrayList<>();
+        for(CourseSection courseSection : courseSections) {
+            for (Module module : courseSection.getModules()) {
+                if(module.getContents()!=null) {
+                    contents.addAll(module.getContents());
+                }
+            }
+        }
+
+        Set<Content> set = new HashSet<>();
+        for(int i =0; i<contents.size(); i++) {
+            while(!set.add(contents.get(i))) {
+                changeName(contents.get(i));
+            }
+        }
+
+        return courseSections;
+    }
+
+    private void changeName(Content content) {
+        String fileName = content.getFilename();
+        String newFileName = fileName;
+        if(!(fileName.lastIndexOf('.')==-1 && fileName.lastIndexOf(')')==-1)) {
```

Comment:
Please add comments for this condition and spacings after `==`. Make sure that you don't cross 100 characters per line.